### DISCUSSION TITLE
Allow to find "Sub-"POIs mapped as tag of a Parent POI (i.e. `bin=yes`)

### DISCUSSION THREAD

**Discussion by bentolor** [organicmaps/organicmaps#1351] [Ideas]
Based on [my limited activities in this project so far](https://github.com/organicmaps/organicmaps/pull/1228/files), _I do not want to give the wrong impression that I somehow have a fixation on :wastebasket: trash cans_ or the like :laughing:  (ok - when you are out with the :bus:  motorhome, you suddenly find that very exciting): But I think my feature request or issue would be probably beneficial in other areas, too.

### How Open Street Map handles POIs like waste bins & others

There are [two common ways to map a waste bin](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dwaste_basket):

1. As explicit POI with `amenity=waste_basket`  
    → [508 641 hits in OSM](https://taginfo.openstreetmap.org/tags/amenity=waste_basket) 
2. Or by [an additional tag `bin=yes`](https://wiki.openstreetmap.org/wiki/Key:bin) as part of  facilities like `bench=*`, `shelter=*`, or `public_transport=platform`.     
   → [239 717 hits in OSM](https://taginfo.openstreetmap.org/keys/bin)

### My issue with Organic Maps

If you search for `waste bin` OrganicMaps will find POIs matching 1. but no POIs matching the mapping pattern 2.

### Related / Similar use cases

I think in the following areas this "Sub-Tag" pattern is also applied and I assume there are a few more cases, too:

* **Find "free" parking slots** as described in #1064, which is focused on the OSM Tag [`fee=no`]( https://taginfo.openstreetmap.org/keys/fee) (1 002 335 hits)
* **[Find Food / Drinks vending machines](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dvending_machine)** mapped as  Sub-Tag `vending=food`, `vending=drinks`, `vending=milk` etc. for `amenity=vending_machine` POIs

### Proposal

Present these case the same way as the point-mapped case. I guess this must be somehow done in the preprocessing step used to create the OrganicMaps app data source files. Can anbody point me where to look? 

Would this be *easily feasible*? What do others think?

**Comment by biodranik** [comment]
How would you display it on the map on other points/ways/relations features? Or is it for search-only?
@vng are there any special hacks to index it for search only, and avoid adding a feature on the map?

**Reply by bentolor** [reply]
Excellent question, @biodranik : 
- **In the first place its really about discovery using the search.** For my use-case of trash bins I'd assume a local user will understand: _Ah! This search result is a bus, stop. Obviously this means the bus stop has an accompanying trash bin!_
- I'd say that `bin=yes` at a point clearly states, that there _is_ a physical trash bin _at that point_. Therefore in my opinion it would'nt be _wrong_ to render this as feature on the  map for point locations.  But what about relations/ways? Or other use cases?

So overall for simplicity and to stick as closely as possible to the _actually mapped data_ I'd say returning them as search results might be the preferable solution, unless it's significant easier to add them as explicit features. 

BTW: If feel this very related to  #374: On doing a a search, OM should not omit direct hits in near proximity depending on the zoom level. Typically you are searching for the _"next hit in my proximity"_ or "all hits in this area to modify my route accordingly".

**Reply by biodranik** [reply]
How would you draw a bin=yes at a bus stop point node? One icon under another?

**Reply by bentolor** [reply]
Yes – finding a good answer here is hard; that's why I'd go for the search-only approach at first.

My 2ct regarding your question:
* from a data point of view, _both_ entities _are_ at the same location. 
* IMHO the common approach to render these _map marker clusters_ is
     * a)  to render clusters differently (i.e with a counter) 
![grafik](https://user-images.githubusercontent.com/588260/135745879-e538bf89-ac1e-4ea4-a816-8213083c9aff.png)
     * and b) expand the cluster indicator on click, attaching the overlapping icons connected by lines to a central dot (illustration; didn't find a concrete example) 
![grafik](https://user-images.githubusercontent.com/588260/135746004-4376ba1f-26e5-4f3c-b52d-199eae94fd9f.png)
* A poor-mans solution could just to a little offset in the range of centimeters on identical feature locations

But in the first step I'd relly  focus on the search for now, which would circumvent this challenge and expensive visualisation.
